src/app/participant/models.py

```diff
@@ -0,0 +1,59 @@
+from app.models import BaseModel
+from google.appengine.ext import ndb
+
+import tinyid
+from app.league.models import LeagueModel
+from app.user.models import UserModel
+
+__author__ = 'Alex'
+
+
+class ParticipantModel(BaseModel):
+    participant_id = ndb.StringProperty(required=True)
+    league_id = ndb.StringProperty(required=True)
+    user_id = ndb.StringProperty(required=True)
+    name = ndb.StringProperty(indexed=False)
+    rating = ndb.FloatProperty()
+
+    @classmethod
+    def generate_id(cls):
+        """ Generate a unique id. """
+        if ndb.in_transaction():
+            tiny_id = tinyid.TinyIDGenerator(namespace='P').generate_tinyid(run_in_transaction=False).upper()
+        else:
+            tiny_id = tinyid.TinyIDGenerator(namespace='P').generate_tinyid(run_in_transaction=True).upper()
+        return '%s-%s' % ('P', tiny_id)
+
+    @classmethod
+    def build_key(cls, participant_id):
+        """ Builds a key in the default namespace. """
+        key = ndb.model.Key(cls.__name__, participant_id)
+        return key
+
+    @classmethod
+    def _post_delete_hook(cls, key, future):
+        participant = key.get()
+        user_key = UserModel.build_key(cls.user_id)
+        league_key = LeagueModel.build_key(participant.league_id, user_key)
+        league = league_key.get()
+        league.update_participant_count(-1)
```

Comment:
so one thing that won't work about this is that you're calling ``` update_participant_count ``` but you're not calling ``` league.put() ``` inside or outside of that function, so your changes will never be saved to datastore